pydantic_ai_slim/pydantic_ai/_output.py

```diff
@@ -83,13 +90,18 @@ class OutputSchema(Generic[OutputDataT]):
     Similar to `Tool` but for the final output of running an agent.
     """
 
+    # TODO: Since this is currently called "preferred", models that don't have structured output implemented yet ignore it and use tools (except for Mistral).
+    # We should likely raise an error if an unsupported mode is used, _and_ allow the model to pick its own preferred mode if none is forced.
+    preferred_mode: Literal['tool', 'structured'] | None  # TODO: Add mode for manual JSON
+    type_adapter: TypeAdapter[OutputDataT]
     tools: dict[str, OutputSchemaTool[OutputDataT]]
-    allow_text_output: bool
+    allow_text_output: bool  # TODO: Verify structured output works correctly with string as a union member
+    json_schema: ObjectJsonSchema  # TODO: Verify structured output works correctly with a union
 
     @classmethod
     def build(
         cls: type[OutputSchema[T]],
-        output_type: type[T] | ToolOutput[T],
+        output_type: type[T] | ToolOutput[T] | StructuredOutput[T],  # TODO: Support a list of output types/markers
```

Comment:
Not super important, but I'll note that I think we only need (aspire?) to support `list[ToolOutput[T]]` actually, not arbitrary lists of output markers.

Well, at least, I see some tangible value in supporting `list[ToolOutput[T]]`, but I don't currently see a very compelling use case for supporting `list[StructuredOutput[T]]` or `list[StructuredOutput[T] | ToolOutput[T]]` or similar.